src/routes/safe/components/Apps/components/ConfirmTxModal/ReviewConfirm.tsx

```diff
@@ -120,11 +121,13 @@ export const ReviewConfirm = ({
     safeTxGas: manualSafeTxGas,
     manualGasPrice,
     manualGasLimit,
+    manualSafeNonce,
   })
 
   const [buttonStatus, setButtonStatus] = useEstimationStatus(txEstimationExecutionStatus)
   const [executionApproved, setExecutionApproved] = useState<boolean>(true)
-  const doExecute = isExecution && executionApproved
+  const canTxExecute = useCanTxExecute(false, manualSafeNonce)
+  const isExecution = canTxExecute && executionApproved
```

Comment:
`executionApproved` sounds a bit ambiguous because transactions can also be approved. Since it is about the checkbox state something like `shouldExecute` would be more fitting. What do you think?

Reply:
This actually makes most sense for the checkbox state. My suggestions, expanded upon my comment above would be:

- `canTxExecute` for hook return
- `shouldExecute` for checkbox state
- `willExecute` for instances where the above two are combined

We can create an issue to update the other occurances of these variables when they appear elsewhere.